### PR TITLE
Admin Place Order not working as expected.

### LINKED ISSUE #31786: Magento 2. Admin Place Order not working as expected.
### Preconditions (*)
1. Magento 2.4.1
1. Not reproduced on 2.4-develop;
1. Reproduced on 2.4-develop + MSI (Inventory);

### Steps to reproduce (*)
1. Create customer with **different** default billing and default shipping address (2 different addresses in total)
1. Go to Admin > Sales > Orders > Create Order 
1. Select customer from step 1
1. Edit any billing input

### Expected result (*)

1. Only billing inputs change

### Actual result (*)

1. Shipping counterpart input changes

## 
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- < ] Severity: **S0** *- Affects critical data or functionality and leaves users without workaround.*
- [x] Severity: **S1** *- Affects critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S3** *- Affects non-critical data or functionality and does not force users to employ a workaround.*
- [ > Severity: **S4** *- Affects aesthetics, professional look and feel, “quality” or “usability”.*

### PR DESCRIPTION
**Pull request by engcom-Kilo**
### Related Pull Requests
https://github.com/magento-commerce/inventory/pull/62

### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/31786

### Manual testing scenarios (*)

1. see https://github.com/magento/magento2/issues/31786

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Review by zakdma — APPROVED** [review]

**Comment by engcom-Kilo** [comment]
@magento run all tests

File: app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminCreateOrderFormShippingAddressActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * seeInField COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AssertAdminCreateOrderFormShippingAddressActionGroup">
```
Comment (zakdma) [inline app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminCreateOrderFormShippingAddressActionGroup.xml:11]: Please rewrite this AG to support other address fields
Example for billing address assert made by @engcom-Echo
https://github.com/magento/magento2/blob/09aec3d4a855d8cbfd4f1806da6d5eac596e02ca/app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminBillingAddressFieldsOnOrderCreateFormActionGroup.xml

**Review by zakdma — CHANGES_REQUESTED** [review]

File: app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminCreateOrderFormShippingAddressActionGroup.xml (hunk 1 above)
Comment (engcom-Kilo) [reply app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminCreateOrderFormShippingAddressActionGroup.xml:11]: Added additional fields.

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

File: app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminCreateOrderFormShippingAddressActionGroup.xml (hunk 1 above)
Comment (zakdma) [reply app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminCreateOrderFormShippingAddressActionGroup.xml:11]: @nmalevanec what about
```
<argument name="province" type="string" defaultValue=""/>
```

**Comment by engcom-Kilo** [comment]
@magento run all tests

**Comment by engcom-Kilo** [comment]
@magento run all tests

File: app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminCreateOrderFormShippingAddressActionGroup.xml (hunk 1 above)
Comment (engcom-Kilo) [reply app/code/Magento/Sales/Test/Mftf/ActionGroup/AssertAdminCreateOrderFormShippingAddressActionGroup.xml:11]: Added

**Review by zakdma — APPROVED** [review]

**Comment by dmitriyprime** [comment]
@magento I am working on this

**Comment by dmitriyprime** [comment]
:heavy_check_mark: QA passed
Was checked case from Manual testing scenario

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31841`.

**Pull request merged**